File: RiptideCore/Sources/RiptideCore/VolumeTable.swift
public struct SetRange: Equatable, Sendable {
    public let low: Int
    public let high: Int
    public init(_ low: Int, _ high: Int) { self.low = low; self.high = high }
}

public enum VolumeTable {
    /// Weekly set range per muscle per effort (spec §4).
    public static func weeklyRange(for muscle: MuscleGroup, effort: Effort) -> SetRange {
        let (min, opt, max): (SetRange, SetRange, SetRange)
        switch muscle {
        case .chest:      (min, opt, max) = (SetRange(5, 8),   SetRange(10, 14), SetRange(15, 20))
        case .lats:       (min, opt, max) = (SetRange(6, 9),   SetRange(12, 16), SetRange(17, 22))
        case .frontDelts: (min, opt, max) = (SetRange(0, 4),   SetRange(4, 8),   SetRange(10, 12))
        case .sideDelts:  (min, opt, max) = (SetRange(6, 10),  SetRange(12, 18), SetRange(20, 26))
        case .rearDelts:  (min, opt, max) = (SetRange(4, 8),   SetRange(10, 16), SetRange(18, 24))
        case .traps:      (min, opt, max) = (SetRange(4, 8),   SetRange(10, 16), SetRange(17, 24))
        case .quads:      (min, opt, max) = (SetRange(4, 8),   SetRange(9, 14),  SetRange(15, 20))
        case .hamstrings: (min, opt, max) = (SetRange(4, 6),   SetRange(8, 12),  SetRange(13, 18))
        case .calves:     (min, opt, max) = (SetRange(5, 8),   SetRange(10, 16), SetRange(18, 24))
        case .triceps:    (min, opt, max) = (SetRange(4, 8),   SetRange(10, 14), SetRange(16, 20))
        case .biceps:     (min, opt, max) = (SetRange(4, 8),   SetRange(10, 14), SetRange(16, 20))
        case .forearms:   (min, opt, max) = (SetRange(0, 3),   SetRange(4, 8),   SetRange(10, 14))
        case .abs:        (min, opt, max) = (SetRange(3, 6),   SetRange(6, 12),  SetRange(14, 18))
        }
        switch effort {
        case .minimal: return min
        case .optimal: return opt
        case .maximal: return max
        }
    }
}

File: web/src/core/volumeTable.ts
import type { MuscleGroup } from './muscleGroup';
import type { Effort } from './effort';

export interface SetRange {
  low: number;
  high: number;
}

export function r(low: number, high: number): SetRange {
  return { low, high };
}

// Weekly set range per muscle per effort (spec §4). Values copied verbatim from
// RiptideCore/Sources/RiptideCore/VolumeTable.swift.
const TABLE: Record<MuscleGroup, { minimal: SetRange; optimal: SetRange; maximal: SetRange }> = {
  chest:      { minimal: r(5, 8),  optimal: r(10, 14), maximal: r(15, 20) },
  lats:       { minimal: r(6, 9),  optimal: r(12, 16), maximal: r(17, 22) },
  frontDelts: { minimal: r(0, 4),  optimal: r(4, 8),   maximal: r(10, 12) },
  sideDelts:  { minimal: r(6, 10), optimal: r(12, 18), maximal: r(20, 26) },
  rearDelts:  { minimal: r(4, 8),  optimal: r(10, 16), maximal: r(18, 24) },
  traps:      { minimal: r(4, 8),  optimal: r(10, 16), maximal: r(17, 24) },
  quads:      { minimal: r(4, 8),  optimal: r(9, 14),  maximal: r(15, 20) },
  hamstrings: { minimal: r(4, 6),  optimal: r(8, 12),  maximal: r(13, 18) },
  calves:     { minimal: r(5, 8),  optimal: r(10, 16), maximal: r(18, 24) },
  triceps:    { minimal: r(4, 8),  optimal: r(10, 14), maximal: r(16, 20) },
  biceps:     { minimal: r(4, 8),  optimal: r(10, 14), maximal: r(16, 20) },
  forearms:   { minimal: r(0, 3),  optimal: r(4, 8),   maximal: r(10, 14) },
  abs:        { minimal: r(3, 6),  optimal: r(6, 12),  maximal: r(14, 18) },
};

export function weeklyRange(muscle: MuscleGroup, effort: Effort): SetRange {
  return TABLE[muscle][effort];
}
